### PR TITLE
Version 1

### PR DESCRIPTION
**Pull request by nottherealalanturing**
## What?
in this capstone project i created a full-stack application with rails with full authentication and authorization features for managing transactions.

## How?
Authentication was implemented with devise, logged in users are directed to the user dashboard which is only accessible to authorized users.

## Testing?
All routes, models, controllers and views were tested to avoid breaking app.

## Screenshots (optional)
Screen Shot 2022-10-18 at 17 15 57
Screen Shot 2022-10-18 at 17 16 27

Screen Shot 2022-10-18 at 17 16 42

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,96 @@
...
+- Run `git checkout -b your-branch-name`
+- Make your contributions
+- Push your branch up to your forked repository
+- Open a Pull Request with a detailed description to the development(or master if not available) branch of the original project for a review
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
```
Comment (OLIPLICHE) [inline README.md:91]: Please make sure the README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -5,12 +5,24 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
+    <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
```
Comment (OLIPLICHE) [inline app/views/layouts/application.html.erb:8]: - [ ] Please try to fix this to provide the login page.

- 
Capture

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are: :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (OLIPLICHE) [inline app/models/user.rb:3]: Nice job in using devise 💯 💯

File: spec/models/group_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
```
Comment (OLIPLICHE) [inline spec/models/group_spec.rb:1]: - Test 
- 

Capture

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @nottherealalanturing, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider**:
- **This project is using Gitflow. You did your work in Gitflow(feature to dev), but you were not supposed to merge the feature to dev and turn to GitHubflow(dev to main). please don't repeat this mistake next time.** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (nottherealalanturing) [reply app/views/layouts/application.html.erb:8]: You need to initialize the build process by running `./bin/dev` after `bundle install`

Comment (codepantha) [inline app/views/layouts/application.html.erb:8]: - [ ] Even after running the `./bin/dev` command you recommended, the problem the previous reviewer highlighted still persists. As you can see, the error is a result of tailwind.css not being present in the asset pipeline. Please see the error below and try to fix it 🙏 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @nottherealalanturing 👋 , 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good UI design
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (nottherealalanturing) [reply app/views/layouts/application.html.erb:8]: I cloned the project again, and ran the `./bin/dev` command and it fixed it.  @codepantha

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @nottherealalanturing 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes from the previous reviewers
- Great communication with the reviewers

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**